### ISSUE TITLE
Add in `composer-require-checker` as a globally installed tool

### ISSUE THREAD

**Issue by internalsystemerror** [laminas/laminas-ci-matrix-action#131]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

[`composer-require-checker`](https://github.com/maglnet/ComposerRequireChecker) is a useful tool for ensuring that `composer.json` contains all the necessary required packages. It is advised to be used as a standalone tool, or as globally installed, not as a development dependency.

Ideally this should be installed into this action as a PHP tool, however this is dependent on the resolution of https://github.com/laminas/laminas-ci-matrix-action/pull/119.

**Comment by boesing** [comment]
we first should see that we merge the matrix container and CI container into one project so we have aligned releases.
Then we can add the tool to the CI container while providing the job via the matrix container.

I would not expect that we need to run `composer require --global` for that. If we want to have this installed for every CI run, we can add it to `/usr/local/bin` and use it.

Not sure why we need anything from #119, can you elaborate?

**Comment by internalsystemerror** [comment]
Because it's a PHP tool requiring PHP and the matrix runs in JS was my thinking, but I guess the JS just needs to set up a job to run like PHPUnit (which I assume is already running on PHP).

**Comment by boesing** [comment]
imho the matrix does not run any checks. The matrix should generate checks and thus I dont see the point why we want to execute the require checker here

**Comment by Ocramius** [comment]
Indeed, matrix stays as-is, it's just the repositories that get together first

**Comment by internalsystemerror** [comment]
I don't think we do want to do that, I was stating why I (mistakenly) thought #119 would be a blocker, but you're right we can just add it as another check once it's added to the ci container (however its installed).
